src/vm/vm_memory/memory.rs

```diff
@@ -66,44 +78,117 @@ impl Memory {
         self.validate_memory_cell(&MaybeRelocatable::from(key))
     }
 
-    pub fn get<'a, K: 'a>(&self, key: &'a K) -> Result<Option<&MaybeRelocatable>, MemoryError>
+    /// Retrieve a value from memory (either normal or temporary) and apply relocation rules
+    pub fn get<'a, 'b: 'a, K: 'a>(
```

Comment:
Effectively we concluded this one should be `pub(crate)` rather than `pub`. Only `VirtualMachine::get` is part of the pubic interface, but not `Memory::get`.